src/compiler/rtl-array.bas

```diff
@@ -26,20 +26,20 @@
 			6, _
 			{ _
 				( FB_DATATYPE_VOID, FB_PARAMMODE_BYDESC, FALSE ), _
-				( FB_DATATYPE_UINT, FB_PARAMMODE_BYVAL, FALSE ), _
-				( FB_DATATYPE_LONG, FB_PARAMMODE_BYVAL, FALSE ), _
-				( FB_DATATYPE_LONG, FB_PARAMMODE_BYVAL, FALSE ), _
-				( FB_DATATYPE_UINT, FB_PARAMMODE_BYVAL, FALSE ), _
-				( FB_DATATYPE_INVALID, FB_PARAMMODE_VARARG, FALSE ) _
+				( typeSetIsConst( FB_DATATYPE_UINT ), FB_PARAMMODE_BYVAL, FALSE ), _
+				( typeSetIsConst( FB_DATATYPE_LONG ), FB_PARAMMODE_BYVAL, FALSE ), _
+				( typeSetIsConst( FB_DATATYPE_LONG ), FB_PARAMMODE_BYVAL, FALSE ), _
+				( typeSetIsConst( FB_DATATYPE_UINT ), FB_PARAMMODE_BYVAL, FALSE ), _
+				( typeSetIsConst( FB_DATATYPE_INVALID ), FB_PARAMMODE_VARARG, FALSE ) _
 			} _
 		), _
 		/' function fb_ArrayRedimPresvEx cdecl _
 			( _
 				array() as any, _
-				byval elementlen as uinteger, _
-				byval doclear as long, _
-				byval isvarlen as long, _
-				byval dimensions as uinteger, _
+				byval elementlen as const uinteger, _
```

Comment:
I really struggled with this decision exactly because of "byval" being treated same as "byval as const".  I don't know, I may have misguided logic here: for the caller, byval implies const, so it seems more descriptive to explicitly include const in the prototype.  Yet, fbc doesn't (or shouldn't) check for const on byval and C ignores it, so end result is useless.  I have no problem rolling back the changes and just doing the byref/ptr types; maybe that would be clearer having the fb rtl built-ins follow what is actually in the C rtlib public API.